app/code/Magento/Catalog/Model/Product/Option/Validator/DefaultValidator.php

```diff
@@ -159,13 +169,24 @@ protected function isInRange($value, array $range)
     }
 
     /**
-     * Check whether value is not negative
+     * Check whether the value is negative
      *
      * @param string $value
      * @return bool
      */
     protected function isNegative($value)
     {
-        return intval($value) < 0;
+        return $this->localeFormat->getNumber($value) < 0;
+    }
+
+    /**
+     * Check whether the value is a number
+     *
+     * @param string $value
+     * @return bool
+     */
+    protected function isNumber($value)
```

Comment:
So @dverkade looking over this pr again I think we should be making this method public. We are trying to stay away from protected and since these classes are built with inheritance in mind I would suggest that having this method public would allow for a user to create a new validator and change the `isNumber` functionality.

Reply:
Hi @dmanners, thanks for reviewing this. I have changed the function to public.